src/views/login/components/login.js

```diff
@@ -4,16 +4,38 @@ import './login.scss';
 export default class Login extends Component {
 
     static propTypes = {
+        usernameChanged: PropTypes.func.isRequired,
+        passwordChanged: PropTypes.func.isRequired,
+        login: PropTypes.func.isRequired,
+
+        usernameField: PropTypes.string.isRequired,
+        passwordField: PropTypes.string.isRequired,
+        history: PropTypes.object.isRequired,
     };
 
+    handleUsernameChanged = (e) => {
+        const { usernameChanged } = this.props;
+        usernameChanged(e.target.value);
```

Comment:
i just like to separate any logic from the jsx so it doesn't get crowded. but yes, the only thing we should be doing here is getting the values of username and password from the inputs